### PR TITLE
force pnpm usage

### PR DESCRIPTION
**Pull request by benjroy**
This PR uses the `engines` field in `package.json` and the `engine-strict` config in `.npmrc` to force the usage of `pnpm` in this repo.   It will throw an error if `npm` is used.

Please note:  this rewrote the lockfile when `pnpm install` was run again

Note: `packageManager` field in `package.json` is for [corepack](https://nodejs.org/api/corepack.html) usage, which is experimental node feature that has to be opted into.  That could be useful for us to explore longterm.

### CODE REVIEW

**Comment by benjroy** [comment]
I can't test out the storybook preview because I don't have access somewhere:

Clicking the link takes me to vercel auth page:
image

Logging in with github takes me to an account-not-found page:

image


Does a switch need to be flipped somewhere on our end to allow my user access?

**Review by christianvuerings — APPROVED** [review]

**Comment by benjroy** [comment]
Just noticed something weird the the repo DX in here:

I had auto-merge enabled for this PR.  Got an approval from you @christianvuerings , but it couldn't auto-merge because there was a conflict in the lockfile.

I then resolved the conflict in that lockfile and pushed up.  It immediately automerged when conflict was resolved based on your previous approval, ⚠️ _but it merged BEFORE the ci test run was complete_ ⚠️ 

That felt _dangerous_ , and has me wondering if this is something we can tweak in the auto-merge setup or if we need to tick the box in github repo settings to dismiss previous approvals when new commits are added.

🤔 🤔 🤔

**Pull request merged**
